### DISCUSSION TITLE
Nginx-SSL configuration

### DISCUSSION THREAD

**Discussion by Rainer5970** [pretix/pretix#2378] [Q&A] [answered]
The documentation of https://docs.pretix.eu/en/latest/admin/installation/manual_smallscale.html is a little bit outdated.
My pretix installation went well for no ssl and I could reach the web application and played a while.
But when I reconfigured to ssl it broke and delivers responses "400 Bad Request" now :-(

As this is my first touch with nginx I admit I've no idea here.
```
server {
        # Add index.php to the list if you are using PHP
        index index.html index.htm index.nginx-debian.html;

        server_name 0f9ab3dc-8cbf-49ce-8a13-cf95d7835ad9.ma.bw-cloud-instance.org;

    listen [::]:443 ssl ipv6only=on default_server;# managed by Certbot
    listen 443 ssl default_server; # managed by Certbot

    ssl_certificate /etc/letsencrypt/live/xxxxxx-8cbf-49ce-8a13-cf95d7835ad9.ma.bw-cloud-instance.org/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/xxxxxx-8cbf-49ce-8a13-cf95d7835ad9.ma.bw-cloud-instance.org/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

    add_header Referrer-Policy same-origin;
    add_header X-Content-Type-Options nosniff;

    location / {
        proxy_pass http://localhost:8345;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto https;
        proxy_set_header Host $http_host;
    }

    location /media/ {
        alias /var/pretix/data/media/;
        expires 7d;
        access_log off;
    }

    location ^~ /media/cachedfiles {
        deny all;
        return 404;
    }
    location ^~ /media/invoices {
        deny all;
        return 404;
    }

    location /static/ {
        alias /var/pretix/venv/lib/python3.9/site-packages/pretix/static.dist/;
        access_log off;
        expires 365d;
        add_header Cache-Control "public";
    }

}
```

Syslog logs this for every request:
 `Dec 13 17:17:31 pretix gunicorn[32344]: WARNING 2021-12-13 17:17:31,769 django.request log Bad Request: /`

**Comment by raphaelm** [comment, marked as answer]
I don't think the documentation is outdated, what gives you that idea?

Is the 400 error page carrying a purple-ish pretix design or a black-on-white nginx design?

Can you check ``pretix.log`` and ``/var/log/nginx/error.log``, respectively for error messages?

If it's pretix-styled, it's most likely an issue with your Host header, e.g. 0f9ab3dc-8cbf-49ce-8a13-cf95d7835ad9.ma.bw-cloud-instance.org does not match what your `pretix.cfg` sets as `url=`.

**Comment by Rainer5970** [comment]
Hi Raphael,

thanks for the quick reply. "Outdated" meant I was probably upset if I've no idea about nginx and just want to copy blindly a valid configuration without thinking and it was indeed just a warning about the _ssl on_:
```
root@pretix:/etc/pretix# nginx -t
nginx: [warn] the "ssl" directive is deprecated, use the "listen ... ssl" directive instead in /etc/nginx/sites-enabled/default:85
nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
nginx: configuration file /etc/nginx/nginx.conf test is successful
```
In my response headers (_Content-Security-Policy_) I found obviously insufficient references for http://ipv4 of my machine which 
should be clearly more like https://hostname and so I found this configuration (_url_) as you mentioned in _/etc/pretix/pretix.cfg_ 

Now it's working. Thanks.
